modules/machineset-non-guaranteed-instance.adoc

```diff
@@ -0,0 +1,26 @@
+// Module included in the following assemblies:
+//
+// * machine_management/creating-infrastructure-machinesets.adoc ?
+// * machine_management/creating_machinesets/creating-machineset-aws.adoc
+// * machine_management/creating_machinesets/creating-machineset-azure.adoc (4.6)
+// * machine_management/creating_machinesets/creating-machineset-gcp.adoc (4.6)
+
+[id="machineset-non-guaranteed-instance_{context}"]
+= Creating a spot instance
+
+You can create an AWS MachineSet that deploys machines as non-guaranteed, spot instances to save costs compared to on-demand instance prices. A machine is paired 1:1 with an instance on AWS. The MachineSet ensures the correct number of machines are created. If AWS does not have capacity, the Machine Health Checker (MHC) can remove the machine. The MHC also remediates a machine that fails if a request for a spot instance is not satisfied when a machine is created. AWS does not replace a spot instance when the instance is terminated.
+
+[IMPORTANT]
+====
+It is strongly recommended that control-plane machines are not created on spot instances. If a control-plane node is terminated, manual intervention is required to replace the node.
+====
+
+.Procedure
+* Add the following fields under the `providerSpec` field in the MachineSet YAML file:
++
+[source,yaml]
+----
+providerSpec:
+  value:
+    spotMarketOptions: {}
+----
```

Comment:
Should we add a note about the optional `maxPrice` field that can be used to limit the cost of the instance? If we do, we should also note that the default is to be charged up to the on-demand price and that it is strongly recommended not to set an explicit max price